src/realm/obj.cpp

```diff
@@ -931,22 +931,45 @@ Obj::FatPath Obj::get_fat_path() const
     return result;
 }
 
-Obj::Path Obj::get_path() const
+FullPath Obj::get_path() const
 {
-    Path result;
-    bool top_done = false;
-    auto sizer = [&](size_t size) {
-        result.path_from_top.reserve(size);
-    };
-    auto step = [&](const Obj& o2, ColKey col, Mixed idx) -> void {
-        if (!top_done) {
-            top_done = true;
-            result.top_table = o2.get_table()->get_key();
-            result.top_objkey = o2.get_key();
-        }
-        result.path_from_top.push_back({col, idx});
-    };
-    traverse_path(step, sizer);
+    FullPath result;
+    if (m_table->is_embedded()) {
+        REALM_ASSERT(get_backlink_count() == 1);
+        m_table->for_each_backlink_column([&](ColKey col_key) {
```

Comment:
Do we need this since there is only one backlink?

Reply:
There may be several backlink columns - one for each Table that points to this embedded Table

Reply:
We assert that we have only one backlink. So although we have several backlink columns, only one of them must contain a link. We have to go through all to find that one.

Reply:
got it